### PR TITLE
Enhncements: Updated installation instructions, added test script, refactoring and example function

### PR DESCRIPTION
**Pull request by soman24**
Hello! I'm a student working on this project as part of my coursework. I've made several changes and additions based on the requirements. Here's a summary of my changes:

Installation Instructions: I have updated the installation instructions in the index.rst file to provide clearer and more detailed steps for setting up the project on different operating systems.

Test Script: I've added a new test script, "test_search.py", which includes unit tests for a core functionality of the project. These tests cover edge cases and ensure the reliability of the code.

Code Refactoring: As part of my documentation efforts, I refactored some of the code in 'setup.py' for improved readability and adherence to best practices. The changes include renaming variables and adding comments.

Added Function: I've also added a new function, "get_related_artists()", in "artist_recommendations.py". This function can serve as a reference for other contributors and users of the project.

I've tested the changes locally to ensure they work as intended and have updated the documentation accordingly. I'm confident that these updates will enhance the project's usability and maintainability. Please let me know if you have any feedback or suggestions. Thank you!

### CODE REVIEW

File: docs/index.rst [hunk 1]
```diff
@@ -76,32 +76,77 @@ API <https://developer.spotify.com/documentation/web-api/>`_ documentation.
...
+- Install Homebrew: Install Homebrew, a package manager for macOS.
+
+- Run the followng command::
+
+   /bin/bash -c "$(curl -fsSL https://raw.githubusercontent.com/Homebrew/install/HEAD/install.sh)"
+
+- Install Python: Run brew install python in the terminal::
+
+   brew install python
+
```
Comment (stephanebruckert) [inline docs/index.rst:102]: I appreciate beginners can use this package, but any python library shouldn't have to explain how to install python, it's assumed that it's a prerequisite

File: tests/unit/test_search.py [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+from spotipy.oauth2 import SpotifyClientCredentials
+import spotipy
+import traceback
+
+
+class artist_info_test_case(unittest.TestCase):
+    def test_artist_info_with_search_str(self):
+        search_str = 'Radiohead'
+        sp = spotipy.Spotify(client_credentials_manager=SpotifyClientCredentials(
+            client_id='id_client', client_secret='secret_client'))
```
Comment (stephanebruckert) [inline tests/unit/test_search.py:11]: I doubt this works with incorrect client id and secret. Even if it would work, requests in unit tests should be mocked. Real network requests can be made in integration tests however.

File: setup.py [hunk 3]
```diff
@@ -1,32 +1,37 @@
...
+    "doc": doc_requirements,
+    "test": test_requirements
 }
 
+# Configure the setup function with required and optional parameters
 setup(
-    name='spotipy',
-    version='2.23.0',
-    description='A light weight Python library for the Spotify Web API',
+    name="spotipy",
```
Comment (stephanebruckert) [inline setup.py:25]: We definitely need consistency with single and double quotes, but this should be managed on the whole project at the linter level. Feel free to open a PR that does only that.
